### PR TITLE
[Bexley] Associate Whitespace missed collection report with correct UPRN

### PR DESCRIPTION
**Pull request by chrismytton**
Change to ensure missed collection reports are always made against the UPRN where the service/collection was found.

Fixes https://github.com/mysociety/societyworks/issues/4439

### CODE REVIEW

File: perllib/FixMyStreet/Cobrand/Bexley/Waste.pm [hunk 1]
```diff
@@ -193,14 +193,17 @@ sub bin_services_for_address {
     my $self = shift;
     my $property = shift;
 
-    my $site_services = $self->whitespace->GetSiteCollections($property->{uprn});
+    my @site_services;
```
Comment (dracos) [inline perllib/FixMyStreet/Cobrand/Bexley/Waste.pm:196]: Looks like this works fine, but if I'm reading it right - you either have the child services, or if they're empty, the parent services, you never have both. So I think you didn't need to add the UPRN to every result, you only needed to note either the child or the parent UPRN depending on whether the child had services or not. Guess up to you if you want to simplify it or keep it as is.

File: perllib/FixMyStreet/Cobrand/Bexley/Waste.pm [hunk 2]
```diff
@@ -435,57 +439,58 @@ sub _remove_service_if_assisted_exists {
...
+                    $ws->{WorksheetStartDate} eq WHITESPACE_UNDEF_DATE ?
+                    '' : $ws->{WorksheetStartDate}
+                ),
+                will_be_completed => (
+                    $ws->{WorksheetEscallatedDate} eq WHITESPACE_UNDEF_DATE ?
+                    '' : $ws->{WorksheetEscallatedDate}
+                ),
+                latest_comment =>
+                    ( $latest_comment ? $latest_comment->text : '' ),
+                uprn => $uprn,
```
Comment (dracos) [inline perllib/FixMyStreet/Cobrand/Bexley/Waste.pm:489]: To just check I understand this - I see this is now looping through both UPRNs if needed, but the UPRN stored here isn't actually used anywhere?

**Review by dracos — COMMENTED** [review]
I think the logic of the first commit can be simplified if you wish to, looks fine otherwise

File: perllib/FixMyStreet/Cobrand/Bexley/Waste.pm (hunk 1 above)
Comment (chrismytton) [reply perllib/FixMyStreet/Cobrand/Bexley/Waste.pm]: Good point! Have added a fixup commit to simplify.

File: perllib/FixMyStreet/Cobrand/Bexley/Waste.pm (hunk 2 above)
Comment (chrismytton) [reply perllib/FixMyStreet/Cobrand/Bexley/Waste.pm]: Ah yeah that's not needed, have removed in a fixup.

**Review by dracos — APPROVED** [review]

**Pull request merged**